corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -170,6 +171,10 @@ hqDefine("data_dictionary/js/data_dictionary", [
         });
 
         self.confirmDeleteProperty = function () {
+            if (!self.id) {
```

Comment:
Just raising the fact that we wouldn't ask users if they are sure they want to delete the property. Personally I think we should still ask users if they are sure, since it's what we do with other properties, so the workflow is the same (we don't surprise users). We could leave out the part mentioning that there's no associated cases/data with this property (or leave it in, because it's true). Curious what others think.

Reply:
Comparing the two options, I agree that it might be better to keep things consistent and show a warning for both scenarios. I would opt to not conditionally remove parts of the modal however, especially if the text remains technically true for both cases. This can just help keep things more consistent.

@mkangia I'm curious to hear your thoughts on the recommended approach since you have also reviewed this code?

Reply:
I'd vote for the option that avoids adding code that isn't worth it.

How bad is it if the user sees the modal for a new property?

Reply:
> How bad is it if the user sees the modal for a new property?

It wouldn't be bad as a user would simply receive a confirmation popup stating that the property is safe to delete. Apart from keeping things consistent, this should also help prevent accidental deletions.

Given the above, I'll remove the code that hides the modal for newly created case properties.

Reply:
[12423cb](https://github.com/dimagi/commcare-hq/pull/34073/commits/12423cb2b07625b09c52b8f808df582540e1f084)